src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -471,8 +471,8 @@ private int execMatlabCommand(Run<?, ?> build, FilePath workspace, Launcher laun
                     + getTestRunTypeList().getBooleanByName("taCoberturaChkBx")
                     + ")),catch e,disp(getReport(e,'extended')),exit(1),end";
         } else {
-            runCommand = "try," + this.getTestRunTypeList().getStringByName("customMatlabCommand")
-                    + ",catch e,disp(getReport(e,'extended')),exit(1),end,exit";
+            runCommand = "try,eval(\"" + this.getTestRunTypeList().getStringByName("customMatlabCommand").replaceAll("\"","\"\"")
```

Comment:
I don't think we have resolved Joe's concerns about security on this yet. I think it is great to see that this nominally works and may be an option, but I am not ready to say ship it until we resolve that conversation

Reply:
I also would like draw your attention to the MATLAB's its own ability to [run any shell command using !](https://www.mathworks.com/help/matlab/matlab_external/run-external-commands-scripts-and-programs.html) . If we allow custom command and user chooses to execute the shell command using this MATLAB feature we will still have the same concerns.

Reply:
@jwilkick-mw . Would it be a good idea if we could try to reproduce the issues on jenkins with above implementation . I shall share .hpi file and a Jenkins instance with this implementation.

Reply:
I played around with it a little bit.  With the input string ") disp('hello') eval(", I got:

ans=
  " ) disp('hello') eval("

which seems pretty reasonable.  I played around with various combinations of using backslashes and even unicode escapes and the closest I came to breaking it was the input:  \")disp('hello')eval("\"   but MATLAB just caught that as an error.  I think this fix might be enough to mitigate breaking out; the whole command is single-quoted as well, so the shell shouldn't try to interpret it.  

The shell MATLAB command does still seem to be a possible concern. Could we have the plugin look for the ! in the custom MATLAB command and refuse to run the command if it finds it?  How does MATLAB Online handle the ! operator?